lib/functions.php

```diff
@@ -441,6 +441,34 @@ function some(array $promises, int $required = 1): Promise
 
         return $result;
     }
+
+    /**
+     * Wraps a promise into another promise, altering the exception or result
+     *
+     * @param Promise $promise
+     * @param callable $callback
+     * @return Promise
+     */
+    function wrap(Promise $promise, callable $callback): Promise
```

Comment:
Could you please adjust this PR to accept a ReactPHP promise like all other functions?

Reply:
done

Reply:
A passed ReactPHP promise needs to be adapted before `onResolve()` can be called.

Reply:
ah totally missed that, my bad :)